src/platform.rs

```diff
@@ -72,6 +72,7 @@ fn cfg_to_expr(cfg: &CfgExpr, platform_var: &str) -> BoolExpr {
             ("target_feature", v) => {
                 Single(format!("builtins.elem {:?} {}Features", v, platform_var,))
             }
+            ("target_has_atomic", _) => True, // TODO: This is required to compile cargo, but how can I determine this in Nix?
```

Comment:
This should be fixed, but can I get this information using functions from nixpkgs?

Reply:
Hi! I don't know the answer myself. I was just perusing the current state of rust builders and came across this PR. I happened to see a pair of commits from @DarkKirb in the cargo2nix network graph that take a shot at addressing the `target_has_atomic` expression:

https://github.com/DarkKirb/cargo2nix/commit/076cd30f8d5314d3f2228e33d01f90dbda71d064
https://github.com/DarkKirb/cargo2nix/commit/29084d950413d317ce5c5901d5f5fd3510a38974

The second one looks like it fixes an issue in the first one. Those are both from the `master` branch in https://github.com/DarkKirb/cargo2nix

Reply:
`rustc -Z unstable-options --print target-spec-json` should output the maximum width of atomic instructions in the field `max-atomic-width`. this would unfortunately require an IFD so maybe this could be precomputed for various targets or read from file in the case a target platform json file is supplied.

the values for default supported targets can be found in this crate: https://github.com/rust-lang/rust/tree/master/compiler/rustc_target/src/spec/targets

the metric i wrote ought to work for many hosted targets.